bootloader/src/pyi_utils.c

```diff
@@ -895,33 +927,51 @@ pyi_utils_create_child(const char *thisfile, const int argc, char *const argv[])
             VS("WARNING: Application is started by systemd socket,"
                "but we can't set proper LISTEN_PID on it.\n");
         }
-        execvp(thisfile, argv_pyi);
+        if (execvp(thisfile, argv_pyi) < 0) {
+            VS("Failed to exec: %s\n", strerror(errno));
```

Comment:
Please check if this is unicode save for all encodings? See #3477

Reply:
I'm not really sure how to handle this, or how to test it. `strerror` returns `char *`, so the solution in #3477 (switching to a function that handles `wchar_t`) doesn't seem applicable.

At a glance, it seems like `printf` (`VS` calls `py_global_printf`) does handle `char *`s with multi-byte encodings. So maybe this just works? 🤔 I'm a bit out of my depth in this particular area.